features/steps/steps/propertysets_qtys_units.py

```diff
@@ -19,6 +22,35 @@
     'QTO_OCCURRENCEDRIVEN': 'IfcObject',
 }
 
+@dataclass
+class ConversionBasedUnitDefinition:
+    """
+    used to hold data from table of conversion-based units defined in the IFC spec
+    Ref: IFC2X3 - https://standards.buildingsmart.org/IFC/RELEASE/IFC4/FINAL/HTML/schema/ifcmeasureresource/lexical/ifcconversionbasedunit.htm
+    Ref: IFC4 - https://standards.buildingsmart.org/IFC/RELEASE/IFC2x3/FINAL/HTML/ifcmeasureresource/lexical/ifcconversionbasedunit.htm
+    Ref: IFC4X3 - https://ifc43-docs.standards.buildingsmart.org/IFC/RELEASE/IFC4x3/HTML/lexical/IfcConversionBasedUnit.htm
+    """
+    Name: str
+    UnitType: str
+    ConversionFactor: str
```

Comment:
I don't know. Is this is the annotation for the constructor or the attribute accessor post construction. For the latter it's wrong. This is of course the nature of python type annotations that it doesn't matter, but I wouldn't do it in this way. I prefer that the type annotations map to the reality in the conceptual model.

And since anyway you have a post_init that does a lot of stuff. I would just create a static from_csv_row() function or something that does the type coercion and then call a clean constructor.